app/models/report.rb

```diff
@@ -130,11 +71,11 @@ def self.expire(conditions = {})
     cond = "reports.created_at < \'#{(Time.now.utc - timerange).to_formatted_s(:db)}\'"
     cond += " and reports.status = #{status}" unless status.nil?
 
-    Log.joins(:report).where(:report_id => Report.where(cond)).delete_all
+    Log.joins(:report).where(:report_id => where(cond)).delete_all
```

Comment:
Seems to have been added in https://github.com/theforeman/foreman/pull/2750/files#diff-2bdfbfb986f7b4ce8a67c89828e71148R99